mindsdb/integrations/handlers/chromadb_handler/chromadb_handler.py

```diff
@@ -0,0 +1,368 @@
+                    k=query.limit.value if query.limit else 5,
+                )
+                result = documents_to_df(collection_data)
+
+            else:
+                # if no 'search_query' provided, run a regular query and filter on metadata if provided
+                collection_data = self._collection.get(
+                    where=where.get("meta_data_filter")
+                )
+                result = pd.DataFrame(collection_data)[
+                    ["ids", "documents", "metadatas"]
+                ]
+
+        if query.limit:
+            # if there is a limit clause, limit the result.
+            result = result.head(query.limit.value)
+
+        return Response(resp_type=RESPONSE_TYPE.TABLE, data_frame=result)
+
+    def run_create_table(self, query: ASTNode) -> Response:
+        """
+        Run a create table query on the ChromaDB database.
+        """
+        collection_name = query.name.parts[-1]
+        self._client.create_collection(collection_name)
+
+        return Response(resp_type=RESPONSE_TYPE.OK)
+
+    def run_insert(self, query):
+        """
+        Run an insert query on the ChromaDB database.
+        """
+        collection_name = query.table.parts[-1]
+        columns = [column.name for column in query.columns]
+        df = pd.DataFrame(data=query.values, columns=columns)
+        documents = split_documents(df, columns)
+
+        # converts list of Documents to embedding vectors and stores them in a ChromaDB collection
+        Chroma.from_documents(
+            documents=documents,
+            embedding=self._embedding_function,
+            persist_directory=self._persist_directory,
+            client_settings=Settings(**self._client_config),
+            collection_name=collection_name,
+        )
+
+        return Response(resp_type=RESPONSE_TYPE.OK)
+
+    def query(self, query: ASTNode) -> Response:
+        """
+        Execute a query on the ChromaDB database.
+        """
+        try:
+            self.connect()
+
+            if isinstance(query, Select):
+                return self.query_collection(query)
+
+            elif isinstance(query, CreateTable):
+                return self.run_create_table(query)
+
+            elif isinstance(query, Insert):
+                return self.run_insert(query)
+
+            else:
+                raise NotImplementedError(
+                    f"Unsupported query type {query.__class__.__name__}!"
+                )
+
+        except Exception as e:
+            log.logger.error(f"Error executing query on ChromaDB client, {e}!")
+            return Response(
+                resp_type=RESPONSE_TYPE.ERROR,
+                error_message=f"Error executing query on ChromaDB client, {e}!",
+            )
+
+    def native_query(self, query: str) -> Response:
+        """
+        Execute a native query on the ChromaDB database.
+        """
+
+        # todo fix this - not sure if we need this?
```

Comment:
Depends on whether we need to support this:

```sql
SELECT * FROM (
    SELECT * FROM ...
)
```

Which I don't think we do? At least not with the scope we've initially agreed on. Perhaps let's replace with a `NotImplementedError` for now, assuming it doesn't break the rest of the handler.